lib/stripe/subscriptions/subscription.ex

```diff
@@ -154,29 +152,44 @@ defmodule Stripe.Subscription do
   @doc """
   Delete a subscription.
 
-  Takes the `id` and an optional map of `params`.
+  Takes the subscription `id` or a `Stripe.Subscription` struct.
   """
-  @spec delete(Stripe.id() | t, Stripe.options()) :: {:ok, t} | {:error, Stripe.Error.t()}
-  def delete(id, opts) when is_list(opts), do: delete(id, %{}, opts)
+  @spec delete(Stripe.id() | t) :: {:ok, t} | {:error, Stripe.Error.t()}
+  def delete(id), do: delete(id, [])
 
-  @spec delete(Stripe.id() | t, params) :: {:ok, t} | {:error, Stripe.Error.t()}
-        when params: %{
-               optional(:at_period_end) => boolean
-             }
-  def delete(id, params) when is_map(params), do: delete(id, params, [])
+  @doc """
+  Delete a subscription.
 
-  @spec delete(Stripe.id() | t, params, Stripe.options()) :: {:ok, t} | {:error, Stripe.Error.t()}
-        when params: %{
-               optional(:at_period_end) => boolean
-             }
-  def delete(id, params \\ %{}, opts \\ []) do
+  Takes the subscription `id` and an optional map of `params`.
+
+  ### Deprecated Usage
+
+  Passing a map with `at_period_end: true` to `Subscription.delete/2`
+  is deprecated.  Use `Subscription.update/2` with
+  `cancel_at_period_end: true` instead.
+  """
+  @deprecated "Use Stripe.Subscription.update/2 with `cancel_at_period_end: true`"
+  @spec delete(Stripe.id() | t, %{at_period_end: true}) :: {:ok, t} | {:error, Stripe.Error.t()}
+  def delete(id, %{at_period_end: true}),
+    do: update(id, %{cancel_at_period_end: true}, [])
```

Comment:
Could leave off the `[]` if you wanted.  Up to you though.

Reply:
True.  Good point.